code/game/objects/structures/props/puzzledoor.dm

```diff
@@ -0,0 +1,92 @@
+// An indestructible blast door that can only be opened once its puzzle requirements are completed.
+
+/obj/machinery/door/blast/puzzle
+	name = "puzzle door"
+	desc = "A large, virtually indestructible door that will not open until some form of requirement(s) is or are met."
```

Comment:
`will not open unless certain requirements are met.`

Try reading what you have out loud. It's awkward